Create CONTRIBUTING.md (#143)
p2pcommons/sdk-js: CONTRIBUTING.md

# Contributing to @p2pcommons/sdk-js

Welcome! We *love* that you're interested in contributing to p2pcommons :purple_heart:

> We recognize all our contributors following the [all-contributors](https://github.com/all-contributors/all-contributors) specification. Contributions of any kind welcome!

Reviewer: ```suggestion
We recognize all our contributors using the [all-contributors](https://github.com/all-contributors/all-contributors) bot. Contributions of any kind welcome!
```

> Come and join us on [Slack](https://libscie.slack.com/) to chat with our team and stay up to date with all [Liberate Science](https://github.com/libscie) goings-on.

Reviewer: ```suggestion
Come and join us on [Slack](https://join.slack.com/t/libscie/shared_invite/zt-9l0ig1x1-Sxjun7D6056cOUQ2Ai_Bkw) to chat with our team and stay up to date with all [Liberate Science](https://libscie.org) goings-on.
```

## Code of conduct

> Please note we adhere to a [Code of Conduct](https://github.com/hypergraph-xyz/cli/blob/master/CODE_OF_CONDUCT.md) and any contributions not in line with it (*tl;dr* be an empathetic, considerate person) will not be accepted. Please notify [@chartgerink](mailto:chris@libscie.org) if anything happens.

Reviewer 2: can we move the code of conduct to a central place, instead of linking to the cli repo?

Author: I suggest we move it to the website when it's up. @chartgerink - any ideas?

Reviewer 3: another option is to create a repo with code of conduct inside one of the orgs and link everything to that repo

Author: Would you suggest a separate repo or perhaps an @libscie/docs repo with code of conduct, maybe glossary and perhaps even the business plan and manifesto (when that's done)?

Reviewer: I'm used to putting a code of conduct in each repo, but I think it makes sense to make that part of the libscie website. Could you leave it like this and I'll get on that this/next week? :) @juliangruber would that be okay with you too?

## Feature requests & bug reports

Feature requests and bug reports can be submitted through [GitHub issues](https://github.com/p2pcommons/sdk-js/issues). If the request concerns a proposal for a change to the module specifications, please submit it to our [specs repo](https://github.com/p2pcommons/specs/issues).

> Templates are offered for feature requests and bug reports. These are not mandatory, but preferred.

Reviewer: ```suggestion
We offer templates for feature requests and bug reports.
```

Active voice :)

### Security vulnerabilities

Please report security vulnerabilities directly to [@chartgerink](mailto:chris@libscie.org).

## Contributing code

### Where to start?

> Our work is organized on our [GitHub Projects board](https://github.com/orgs/p2pcommons/projects/1). You can help us the most by contributing to issues in the "Backlog" column. These are issues that are already specified and considered ready to start work on.

Reviewer 2: I don't think anyone should need to know about our column system. People can simply report issues and open PRs, we will triage them as we see fit. If an outsider wants to work on an issue, we can't assign it to them anyway, unless they get added to the project. Usually people can just comment that they want to work on this.

Reviewer 3: I like those changes

Reviewer 2: Sounds great!

Author: @dpaez @juliangruber Update made!

> The "PM Inbox" column and our [Roadmap](https://github.com/p2pcommons/sdk-js/wiki/Roadmap) contain issues that we are planning on working on further down the line. If you're enthusiastic about one of these features, come and discuss it with us on [Slack](https://libscie.slack.com/).

Reviewer: ```suggestion
The "PM Inbox" column and our [Roadmap](https://github.com/p2pcommons/sdk-js/wiki/Roadmap) contain issues that we are planning on working on further down the line. If you're enthusiastic about one of these features, come and discuss it with us on [Slack](https://libscie.slack.com/) ([invite link](https://join.slack.com/t/libscie/shared_invite/zt-9l0ig1x1-Sxjun7D6056cOUQ2Ai_Bkw)).
```

> Technical improvements, bug fixes, documentation and other non-feature work is also totally welcome. Please create an issue first, but feel free to start work right away.

Reviewer: ```suggestion
Technical improvements, bug fixes, documentation and other non-feature work is also totally welcome.
```

Reviewer 2: Agreed, at this stage _any_ contribution is better than following a process

***Please note that we have [specs](https://github.com/p2pcommons/specs/issues) that the application must adhere to. If you believe these specs need amending, submit an issue to that repo.***

### Git guidelines

> When working on an issue, self-assign and switch the `ready to start` label for `in progress`. Please always create a new branch for your work and submit a pull request that closes the issue when done. If you'd like input along the way, a draft pull request is also fine. Please invite recent contributors to review (at least one, but preferably two or more). Project maintainers will take care of releases.

Reviewer: This is only possible for members of the repository. 

Repository maintainers should sign their commits with their PGP key.

Author: Are you referring to the labels or also the branching & PR? Would it make sense to use forks instead @dpaez @juliangruber ?

Reviewer 3: external contributors will need to fork to contribute. We can help them with labels if needed.

Author: @dpaez Update made! Please make or suggest changes to the forking stuff if necessary :)

Reviewer 3: updated

Author: Nice!
